remember.go

```diff
@@ -0,0 +1,97 @@
+package remember
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"strings"
+)
+
+type Memory struct {
+	input []io.Reader
+	output io.ReadWriter
+}
+
+type Option func(*Memory) error
+
+func WithInput(args []string) Option {
+	return func(m *Memory) error {
+		var input = []io.Reader{}
+		for _, str := range args {
+			input = append(input, bytes.NewBufferString(str))
+		}
+		m.input = input
+		return nil
+	}
+}
+
+func WithOutput(output string) Option {
+	return func(m *Memory) error {
+		m.output = bytes.NewBufferString(output)
+		return nil
+	}
+}
+
+func NewMemory(opts ...Option) (Memory, error) {
+	_, err := os.Stat("store.txt")
+	if err != nil {
+		os.Create("store.txt")
+	}
+	defaultFile, err := os.OpenFile("store.txt", os.O_APPEND|os.O_RDWR, 0644)
```

Comment:
This is a nit-pick: the variable name `defaultFile` makes me think there is (or will be) a non-default instance elsewhere. I'm not sure `default` should be part of the name, but there may be a purpose I don't yet understand.

Reply:
Now that I look at this with fresh eyes, I think my `NewMemory` function is overloaded, I do two things in it :|
I open the data store and I create the memory hrmm